lib/ace/transport_app.rb

```diff
@@ -28,6 +29,61 @@ def initialize(config = nil)
       super(nil)
     end
 
+    def self.init_puppet_target(device_json)
+      begin
+        device = JSON.parse(device_json)
+      rescue StandardError
+        # Resource JSON error
+        # Do not add error to log as potentially includes passwords
+        ACE::Error.new('There was an error parsing the json of the Puppet target',
+                       'ace/puppet-target-error')
+        return nil
+      end
```

Comment:
Looking at https://github.com/puppetlabs/ace/blob/5019cabcaf1eb9d0242792df09c4cbfb97a971e1/lib/ace/transport_app.rb#L92-L93, the incoming JSON will already be parsed and validated at this point. Saves you some code here.

Reply:
Removed

Reply:
github messed up the anchoring of this comment.